test/unit/middleware/prepareQuery.js

```diff
@@ -29,16 +26,17 @@ describe("prepareQuery", () => {
 
       options.allowRegex = false;
 
-      prepareQuery(options)(req, {}, next);
-
-      assert.deepEqual(req.erm.query, {
-        query: {
-          foo: {},
-        },
-      });
-      sinon.assert.calledOnce(next);
-      sinon.assert.calledWithExactly(next);
-      sinon.assert.notCalled(options.onError);
+      getPrepareQueryHandler(options)(req, {}, next);
+
+      sinon.assert.calledOnce(options.onError);
```

Comment:
⚠️  When `allowRegex` is `false`, an error will be returned instead of silently removing that part of the query.